R/class-network.R

```diff
@@ -0,0 +1,48 @@
+# Network class
+check_network <- function(object) {
+
+  errors <- character()
+
+  # Check that all nodes in links are in nodes
+  nodesInLinks <- NodeList(union(getSourceNodes(object@links), getTargetNodes(object@links))) # may become a method later if i find i use it elsewhere
+  if (!all(getNodeIds(nodesInLinks) %in% getNodeIds(object@nodes))) {
+    errors <- c(errors, 'Found a node in a link that is not in the node list. All nodes in links must also be in nodes.')
+  }
+
+  # Check that linkColorScheme is one of the accepted values
+  if (!object@linkColorScheme %in% c('none', 'posneg')) {
+    errors <- c(errors, 'linkColorScheme must be one of "none" or "posneg"')
+  }
+
+
+  return(if (length(errors) == 0) TRUE else errors)
+}
+
+
+#' Network
+#' 
+#' A class for representing networks. A network is composed of nodes and links (edges, connections, etc.). A link is represented
+#' as a pair of nodes, with optional attributes such as weight (see Link). To represent a network, we need both the list of links in the network and a list of nodes
+#' in case some nodes have no links. A network can also have properties such as directedness, levels, colors, etc. (coming soon).
+#' 
+#' @slot links LinkList object defining the links in the network.
+#' @slot nodes NodeList object defining the nodes in the network. Some nodes may not have any links.
+#' @slot linkColorScheme string defining the type of coloring scheme the links follow. Options are 'none' (default) and 'posneg'.
```

Comment:
Posneg ? Is this at risk of being specific to correlation? What other options might we see here eventually, for clarity? Do we need a node color scheme?

Reply:
Posneg would also be good for other networks, like differential networks that have +/- some change value associated with each link. But some other options we will likely use are sequential, diverging, or avg degree of parent nodes. And yes node color schemes would be implemented the same way when we need them. Likely node color schemes are by degree, by weighted degree, by centrality, by yourFavoriteNetworkMetric, by layer

Reply:
See my other comments about making color part of method params maybe